app/src/main/java/org/simple/clinic/summary/PatientSummaryScreenController.kt

```diff
@@ -450,15 +453,46 @@ class PatientSummaryScreenController @Inject constructor(
   }
 
   private fun showUpdatePhoneDialogIfRequired(events: Observable<UiEvent>): Observable<UiChange> {
-    val patientUuidStream = events
-        .ofType<PatientSummaryScreenCreated>()
+    val screenCreations = events.ofType<PatientSummaryScreenCreated>()
+
+    val showForInvalidPhone = screenCreations
         .map { it.patientUuid }
+        .switchMap { patientUuid ->
+          hasInvalidPhone(patientUuid)
+              .take(1)
+              .filter { invalid -> invalid }
+              .map { { ui: Ui -> ui.showUpdatePhoneDialog(patientUuid) } }
+        }
 
-    return patientUuidStream
-        .switchMap { patientUuid -> Observables.zip(phoneNumber(patientUuid), lastCancelledAppointment(patientUuid)) }
-        .filter { (number, appointment) -> appointment.updatedAt > number.updatedAt }
-        .take(1)
-        .map { (number) -> { ui: Ui -> ui.showUpdatePhoneDialog(number.patientUuid) } }
+    val showForMissingPhone = screenCreations
```

Comment:
Wasn't the trigger for this supposed to be when a new BP is entered? I don't see this description having changed on Zeplin.

Reply:
Whoops. I got very confused yesterday and forgot to think about this case. Fixing this.